apps/aevm/src/aevm_ae_primops.erl

```diff
@@ -176,109 +176,123 @@ oracle_ttl_t() ->
 sign_t() ->
     {tuple, [word, word]}.
 
-oracle_call_register(_Value, Data, State) ->
+oracle_call_register(_Value, Data, #chain{api = API, state = State} = Chain) ->
     ArgumentTypes = [word, sign_t(), word, oracle_ttl_t(), typerep, typerep],
-    [Acct, Sign0, QFee, TTL, QType, RType] = get_args(ArgumentTypes, Data),
-    case chain_ttl_delta(TTL, State) of
+    [Acct, Sign0, QFee, TTL, QFormat, RFormat] = get_args(ArgumentTypes, Data),
+    %% The aeo_register_tx expects the formats to be binary, althought, atoms are provided.
+    %% This should be solved in a different ticket.
+    case chain_ttl_delta(TTL, Chain) of
         {error, _} = Err -> Err;
         {ok, DeltaTTL = {delta, _}} ->
+            {ok, Tx} = API:oracle_register_tx(<<Acct:256>>, QFee, TTL, QFormat, RFormat, State),
             Callback =
-                fun(API, ChainState) ->
-                    case API:oracle_register(<<Acct:256>>, to_sign(Sign0), QFee, TTL, QType, RType, ChainState) of
+                fun(ChainAPI, ChainState) ->
+                    case ChainAPI:oracle_register(Tx, to_sign(Sign0), ChainState) of
                         {ok, <<OKey:256>>, ChainState1} -> {ok, OKey, ChainState1};
                         {error, _} = Err                -> Err
                     end
                 end,
-            DynCost = state_gas_cost(oracle_registration, DeltaTTL),
-            ?TEST_LOG("~s computed gas cost ~p from relative TTL ~p", [?FUNCTION_NAME, DynCost, DeltaTTL]),
-            {ok, DynCost, fun() -> call_chain(Callback, State) end}
+            SizeGas = size_gas(Tx),
+            StateGas = state_gas(oracle_registration, DeltaTTL),
```

Comment:
The size of what we store in the state depends on the size of the query/response descriptions/types?!

Reply:
Maybe overkill to consider it here?